pkg/resource/deploy/plan_executor.go

```diff
@@ -197,18 +198,18 @@ func (pe *planExecutor) Execute(callerCtx context.Context, opts Options, preview
 		return res
 	}
 
-	var err error
-	if res != nil {
-		err = res.Error()
-	}
-
 	// Figure out if execution failed and why. Step generation and execution errors trump cancellation.
-	if err != nil || pe.stepExec.Errored() {
-		err = execError("failed", preview)
+	if res != nil || pe.stepExec.Errored() {
+		// TODO(cyrusn): We seem to be losing any information about the original 'res's errors.  Should
+		// we be doing a merge here?
+		pe.reportExecResult("failed", preview)
+		return result.Bail()
```

Comment:
importantly, we report, then bail. We want 'error results' to mean "a true 'error' error happened" as opposed to "something normal happened, we reported it and can now quit"